src/httpClient.ts

```diff
@@ -41,79 +52,130 @@ export default class httpClient {
+            // Convert response to AirtableRecord type
+            const record: AirtableRecord = {
+                include_in_n: true,
+                source_name: item.source_name,
+                lead_org: item.lead_org,
+                first_author: item.first_author,
+                source_type: item.source_type,
+                study_status: item.study_status,
+                study_type: item.study_type,
+                test_type: item.study_type,
+                specimen_type: Array.isArray(item.specimen_type) ? item.specimen_type : [item.specimen_type],
+                isotypes_reported: item.isotypes_reported,
+                sex: item.sex,
+                approving_regulator: item.approving_regulator,
+                country: item.country,
+                state: item.state,
+                city: item.city,
+                population_group: item.population_group,
+                age: item.age,
+                denominator: item.denominator_value,
+                seroprevalence: item.serum_pos_prevalence,
+                sample_size: null,
+                sampling_start_date: null,
+                sampling_end_date: item.sampling_end_date,
+                overall_risk_of_bias: item.overall_risk_of_bias,
+                estimate_grade: item.estimate_grade
+            };
+
+            return record;
+        });
+        return records
+    }
+
+    async getAllFilterOptions() {
+       // Fill this with Austin's new endpoint
     }
+    
+    async getAirtableRecords(filters: Filters,
+        sorting_key = "denominator_value",
+        reverse = false) {
+        const reqBodyFilters: Record<string, string[]> = {}
+
+        Object.keys(filters).forEach((o: string) => {
+            const filter = Array.from(filters[o as FilterType]);
+            if (o !== 'publish_date') {
+                reqBodyFilters[o] = filter as string[]
+            }
+        });
 
-    async getAirtableRecords() {
-        const response = await this.httpGet('/airtable_scraper/records')
-        if (!response || !("records" in response) || !("updated_at" in response)) {
+        const date = filters['publish_date'] as Array<Date>
+        const [startDate, endDate] = this.formatDates(date)
+        const reqBody = {
+            filters: reqBodyFilters,
+            start_date: startDate,
+            end_date: endDate,
+            sorting_key: sorting_key,
+            reverse: reverse,
+            per_page: null,
+            page_index: null,
+            columns: [],
+        }
+        const response = await this.httpPost('/data_provider/records', reqBody)
+        if (!response) {
             return [];
         }
-        const airtable_records = response.records!.map((item: Record<string, any>) => {
+        const filtered_records = response.map((item: Record<string, any>) => {
             // Convert response to AirtableRecord type
             const record: AirtableRecord = {
```

Comment:
suggestion: have some sort of function to return this big airtable record object